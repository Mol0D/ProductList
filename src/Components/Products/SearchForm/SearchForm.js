import React from 'react';
import {Container, Row, Col, Form, Button} from 'react-bootstrap';
import history from '../../App/history';
import PropTypes from 'prop-types';

class SearchForm extends React.Component{

  constructor (props) {
    super(props);
    this.state = {
        inputSearch: this.props.search
    }
  }
  
  handleReset = () => {
    this.setState({inputSearch: ''})
  }

  handleSubmit = event => {
    event.preventDefault();
    if(this.props.filter === undefined){
      history.push({pathname: `/products`, search: `?name=${this.state.inputSearch}`})
    } else {
      history.push({pathname: `/products/${this.props.filter}`, search: `?name=${this.state.inputSearch}`})
    }
  }

  handleChange = event =>{
    this.setState({inputSearch: event.target.value})
  }

  render(){
    /* console.log(this.props) */
    return(
      <Form className="form" onSubmit={this.handleSubmit}>
        <Form.Group>
          <Container>
            <Row style={{textAlign: 'center'}}>
              <Col xs="10">
              <Form.Control type="text" 
                placeholder="Search" 
                value={this.state.inputSearch}
                onChange={this.handleChange}
              />
              </Col>
              <Col xs="2">
                <Button variant="danger" type="reset" onClick={this.handleReset}>Reset</Button>
              </Col>
            </Row>
          </Container>
        </Form.Group>
      </Form>
    )
  }   
}

SearchForm.propTypes = {
  filter: PropTypes.string
}

export default SearchForm;