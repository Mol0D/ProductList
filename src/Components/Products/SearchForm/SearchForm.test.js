import React from 'react';
import {shallow, mount} from 'enzyme';
import SearchForm from './SearchForm';

describe("SearchForm", () => {

    const props = {
        filter: "some filter"
    }

    it("without component", () => {
        const wrapper = shallow(<SearchForm/>);
        expect(wrapper).toMatchSnapshot();
    })

    it("with component", () => {
        const wrapper = shallow(<SearchForm {...props}/>);
        expect(wrapper).toMatchSnapshot()
    })

    it("correct Prop Filter", () => {
        const wrapper = mount(<SearchForm filter={props.filter}/>);
        expect(wrapper.prop("filter")).toEqual("some filter")
    })
}) 