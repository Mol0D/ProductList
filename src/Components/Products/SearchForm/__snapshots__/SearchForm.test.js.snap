// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`SearchForm with component 1`] = `ShallowWrapper {}`;

exports[`SearchForm without component 1`] = `ShallowWrapper {}`;
